Framework/DataObjects/src/EventList.cpp

```diff
@@ -3753,6 +3771,46 @@ void EventList::filterByTimeAtSample(Types::Core::DateAndTime start, Types::Core
   }
 }
 
+/** Filter this EventList into an output EventList, using TimeROI
+ * keeping only events within the >= start and < end pulse times.
+ * Detector IDs and the X axis are copied as well.
+ *
+ * @param timeRoi :: reference to TimeROI to be used for filtering
+ * @param output :: reference to an event list that will be output.
+ * @throws std::invalid_argument If output is a reference to this EventList
+ */
+void EventList::filterByTimeROI(Kernel::TimeROI *timeRoi, EventList &output) const {
+  if (this == &output) {
+    throw std::invalid_argument("In-place filtering is not allowed");
+  }
+
+  sort(this->order);
```

Comment:
This is unnecessary because you iterate over all events for each `SplittingInterval`:
```cpp
    std::copy_if(events.begin(), events.end(), std::back_inserter(output),
                 [start, stop](const T &t) { return (t.m_pulsetime >= start) && (t.m_pulsetime < stop); });
```

Reply:
sort has been removed.